backend/dataall/modules/loader.py

```diff
@@ -0,0 +1,64 @@
+"""Load modules that are specified in the configuration file"""
+import importlib
+import logging
+from enum import Enum
+from typing import List
+
+from dataall.core.config import config
+
+log = logging.getLogger(__name__)
+
+_MODULE_PREFIX = "dataall.modules"
+
+
+class ImportMode(Enum):
+    """Defines importing mode
+
+    Since there are different infrastructure components that requires only part
+    of functionality to be loaded, there should be different loading modes
+
+    Keys represent loading mode while value a suffix im module loading.
+    For example, API will try to load a graphql functionality under a module directory
+    The values represent a submodule and should exist
+    """
+
+    API = "gql"
```

Comment:
[Idea/Question] - Customers usually customize certain APIs from a module. How are we going to handle small-customizations of a module? How do we allow customers to do add configurations to the module? One way to solve this with the current design is by copy/pasting the module and create a new "NotebooksCustom" module. It is a solution, but it can result of many custom" forks" which is what we are trying to avoid. I think it is a little soon to be included in this PR, but Ifor the future I would like to have something in the config.json such as:
- modules:
    - notebooks:
       max_notebooks_per_user: 5
       automatic_stop_instance: true

Reply:
It can be easily achieved with a small changes to the current design:

1) With a change of the structure of config.json and how we read it.
2) Providing a better interface how to get a nested properties like `config.get_property("modules.notebooks.max_notebooks_per_user")`

If we 100% sure that we will need, I can add these changes in this PR

Reply:
I think we should first merge this one and then work on expanding the feature, but niiiiice

Reply:
And in the end it will be hard to integrate those "forks". This is a good way to provide completely new modules but for extending existing there should be more granule extension points so that one can use composition or inheritance and modify only what is needed e.g. add additional field to db/api. If I copy the whole module it will be hard for me to follow upstream changes (even harder the it is now because those changes will be physically in other files). But probably this is smth to address as a next step

Reply:
I thought about your suggestion:
We can even override the way how the modules should be loaded. We can do this in the config file. If there are no overrides, then we use default loading.
It can potentially look like this:
```
{
    "modules": [
         {
                "module": "notebook",
                "enabled": "true",
                "loading": {
                       "API": "new_schema"
                }
                "parameters": {}
         }
    ]
}
```
This will allow us to provide a new entrypoint for an existing module. In the example above, we load `new_schema` submodule instead of `gql`. This allow us to add/delete submodules.`new_schema` can even load `gql` patching some methods/classes/functions. That will make our application more customizable. The small downside of this approach: 1) hard to learn how it works (can be solved with well-written documentation) 2) configuration overhead :)

Reply:
- [x] Added a more clear way to disable module in the config.json
- [x] Changed a little the structure of config.json.
- [x] Changed a way how the modules are loaded. Provided the interface to load module in the code, so it should suffice for the existing modules :)